File: prns-react-native/ios/PrnsHostModule.swift
import ExpoModulesCore
import Foundation

public final class PrnsHostModule: Module {
  public func definition() -> ModuleDefinition {
    Name("PrnsHostPlatform")
    Constants(["nativeImage": PrnsNativeImage.name])
    AsyncFunction("defaultStoragePath") { () throws -> String in
      let support = try FileManager.default.url(
        for: .applicationSupportDirectory, in: .userDomainMask,
        appropriateFor: nil, create: true
      )
      let root = support.appendingPathComponent("prns", isDirectory: true)
      try FileManager.default.createDirectory(at: root, withIntermediateDirectories: true)
      return root.path
    }
  }
}

File: prns-react-native/android/src/main/java/rs/reticulum/prns/expo/PrnsHostModule.kt
package rs.reticulum.prns.expo

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.io.File

class PrnsHostModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("PrnsHostPlatform")
    Constants("nativeImage" to PrnsNativeImage.name)
    AsyncFunction("defaultStoragePath") {
      val context = requireNotNull(appContext.reactContext) { "React context is unavailable" }
      val directory = File(context.applicationContext.filesDir, "prns")
      check(directory.isDirectory || directory.mkdirs()) { "Could not create PRNS storage" }
      directory.absolutePath
    }
  }
}
